File: Problems/Easy/1408_string_matching_array.py
class Solution:
    def stringMatching(self, words: list[str]) -> list[str]:
        # res = []
        # for i in range(0, len(words)):
        #     for j in range(0, len(words)):
        #         if i == j:
        #             continue
        #
        #         if words[j] in words[i] and words[j] not in res:
        #             res.append(words[j])
        #
        # return res

        res = []
        fullStr = " ".join(words)
        for word in words:
            if fullStr.count(word) >= 2:
                res.append(word)

        return res

if __name__ == '__main__':
    print(Solution.stringMatching(Solution, ["leetcode","et","code"]))

File: Swift-Solution/Easy/1408_string_matching_array.swift
//Input: words = ["mass","as","hero","superhero"]
//Output: ["as","hero"]
//Explanation: "as" is substring of "mass" and "hero" is substring of "superhero".
//["hero","as"] is also a valid answer.

class Solution {
    func stringMatching(_ words: [String]) -> [String] {
        var res: Set<String> = []

        for i in 0..<words.count {
            for j in 0..<words.count {
                if words[i] == words[j] { continue }

                if words[i].contains(words[j]) {
                    res.insert(words[j])
                }
            }
        }

        print(res)

        return Array(res)
    }
}

Solution().stringMatching(["ga","ugao","dbh","a"]);